SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DormancyTests/DynamicActorAwakeAfterDormantChangePropertyTest.cpp

```diff
@@ -17,33 +29,35 @@ void ADynamicActorAwakeAfterDormantChangePropertyTest::PrepareTest()
 
 	// Step 1 - Spawn dormancy actor and change NetDormancy to DORM_DormantAll
 	AddStep(TEXT("ServerSpawnDormancyActor"), FWorkerDefinition::Server(1), nullptr, [this]() {
-		AActor* Actor = CreateDormancyTestActor();
-		Actor->SetNetDormancy(DORM_DormantAll);
-		RegisterAutoDestroyActor(Actor);
+		DormancyActor = SpawnActor<ADormancyTestActor>(FActorSpawnParameters(), /*bRegisterAsAutoDestroy*/ true);
+		// Going from DORM_DormInitial to DORM_DormantAll
+		DormancyActor->SetNetDormancy(DORM_DormantAll);
 		FinishStep();
 	});
 
-	// Step 2 - Client check NetDormancy is DORM_DormantAll
-	AddStep(
-		TEXT("ClientRequireDormancyTestState"), FWorkerDefinition::AllClients, nullptr, nullptr,
-		[this](float DeltaTime) {
-			RequireDormancyActorCount(1);
-			RequireDormancyTestState(DORM_DormantAll, /*TestRepProperty*/ 0, /*ActorCount*/ 1);
-			FinishStep();
-		},
-		5.0f);
+	// Step 2 Require channel to actually be dormant
+	AddStep(TEXT("ServerAssertActuallyDormant"), FWorkerDefinition::Server(1), nullptr, nullptr, [this](float) {
+		const TSet<TSharedPtr<FNetworkObjectInfo>, FNetworkObjectKeyFuncs>& Actives = DormancyActor->GetNetDriver()->GetNetworkObjectList().GetActiveObjects();
```

Comment:
This would also be true if the object hasn't been replicated yet. What about checking the DormantObjectOnAllConnections like you had initially?

Reply:
Fair point